fix(circuits): Validate properly min revertible sideffect counter after recent changes

sirasistant opened the pull request:
After this PR https://github.com/AztecProtocol/aztec-packages/pull/18679 side effect arrays are not validated to be locally increasing. A test to check that the min revertible side effect counter didn't split a nested call relied on that ordering. To fix this, we treat the min revertible side effect counter as a sideffect so it can't overlap with any other side effect.

LeilaWang reviewed (approved):
Makes more sense to check everything together!

sirasistant committed:
fix(circuits): Validate properly min revertible sideffect counter after recent changes

After this PR https://github.com/AztecProtocol/aztec-packages/pull/18679 side effect arrays are not validated to be locally increasing. A test to check that the min revertible side effect counter didn't split a nested call relied on that ordering. To fix this, we treat the min revertible side effect counter as a sideffect so it can't overlap with any other side effect.
